### PR TITLE
Support both trusted internal and untrusted cross-world unix domain socket

### PR DESCRIPTION
**Pull request by jessehui**
Required:
- [x] #171 
- [x] #176 
- [x] https://github.com/occlum/sefs/pull/36

### CODE REVIEW

File: src/libos/crates/async-io/src/util/channel.rs [hunk 1]
```diff
@@ -280,9 +418,91 @@ impl Consumer {
...
+                return_errno!(EAGAIN, "no data are present to be received");
+            }
+
+            if poller.is_none() {
+                poller = Some(Poller::new());
+            }
+            let mask = Events::IN;
+            let events = self.pollee().poll(mask, poller.as_mut());
+            if events.is_empty() {
+                poller.as_ref().unwrap().wait().await;
```
Comment (guzongmin) [inline src/libos/crates/async-io/src/util/channel.rs:474]: We should use "poller.as_ref().unwrap().wait().await?" to handle the interrupt.

File: src/libos/src/net/addr.rs [hunk 2]
```diff
@@ -27,8 +30,12 @@ impl AnyAddr {
                 Self::Ipv6(ipv6_addr)
             }
             libc::AF_UNIX | libc::AF_LOCAL => {
-                let unix_addr = UnixAddr::from_c_storage(c_addr, c_addr_len)?;
-                Self::Unix(unix_addr)
+                // TODO: Re-enable untrusted Unix domain socket
```
Comment (guzongmin) [inline src/libos/src/net/addr.rs:33]: Should we handle it in this PR?

Comment (jessehui) [reply src/libos/src/net/addr.rs]: This has been handled. Changed the comments.

File: src/libos/crates/async-io/src/util/channel.rs (hunk 1 above)
Comment (jessehui) [reply src/libos/crates/async-io/src/util/channel.rs]: Got it. Fixed.

File: demos/cross_world_uds/run_cross_world_uds_test.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+new_json="$(jq '.untrusted_unix_socks += [
+    {"host": "/tmp/occlum/test.sock", "libos": "/root/test.sock"},
+    {"host": "/tmp/root/", "libos": "/root/"},
+    {"host": "../test.sock", "libos":"/tmp/test.sock" }]' Occlum.json)" && \
+echo "${new_json}" > Occlum.json
+
+mkdir -p /tmp/occlum
+mkdir -p /tmp/root
+cp ../client image/bin
```
[the remark this answers is not in the document]
Comment (jessehui) [reply demos/cross_world_uds/run_cross_world_uds_test.sh]: Done.

**Review by guzongmin — DISMISSED** [review]
LGTM

**Pull request merged**